pkg/scheduler/apis/config/validation/validation.go

```diff
@@ -216,3 +217,115 @@ func validateExtendedResourceName(name v1.ResourceName) []error {
 	}
 	return validationErrors
 }
+
+// ValidateInterPodAffinityArgs validates that InterPodAffinityArgs are correct.
```

Comment:
Just like we separated pluginarg_types in their own file, we can do the same for this validations.
Just to make code navigation easier.

Reply:
Also, we should be calling these functions from `ValidateKubeSchedulerConfiguration`.
However, I don't know if there is a way to avoid a big switch. Any ideas here @liggitt ?

Reply:
a registry of validation functions by type like `map[reflect.Type]func(interface{}) field.ErrorList` would let you write a unit test that ensured all known Args types had a registered validation function, and in ValidateKubeSchedulerConfiguration you could look up a validation function by arg type and call it if it exists

Reply:
Right, over a simple switch, we can validate that we didn't miss functions.

nit: `map[reflect.Type]func(runtime.Object)`

I don't like that we have to cast again inside the validation functions, but that's due to the lack of generics in go. On the other hand, we can get rid of such checks inside plugins' factory functions (`New`).

Reply:
According to @ahg-g's [comment ](https://github.com/kubernetes/kubernetes/issues/90859#issuecomment-626793416) we still should be calling the validation when `New`'ing the plugins. 

Calling it from `ValidateKubeSchedulerConfiguration`  would cause the validation functions to be called twice.

Reply:
That's ok. Calling from ValidateKubeSchedulerConfiguration as well lets us validate statically before we instantiate a running scheduler

Reply:
Right, it is ok to validate twice for now . This is the canonical place for validation, validating at instantiation is needed because cluster autoscaler instantiates a framework statically, so we need coverage for that path.

Reply:
I will use moved validation functions in `ValidateKubeSchedulerConfiguration` in a separate PR after I move Node Resources plugins validation.

Reply:
sg, please add this to the notes or leave a comment in the issue.